### PR TITLE
Add checkbox button

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR adds basic support for checkbox button. The `Animate` label below is **not part of the button** but is given by [`add_text()`](https://docs.pyvista.org/plotting/plotting.html?highlight=add_text#pyvista.BasePlotter.add_text) instead.

To illustrate (inspired by [Orbiting](https://docs.pyvista.org/examples/02-plot/orbit.html?)):

output

Source code: https://gist.github.com/GuillaumeFavelier/1bfa7ad2160d86b0814ed890dd7f27bc

*I think this demo could be improved with [floors](https://github.com/pyvista/pyvista/pull/423) or [background image](https://github.com/pyvista/pyvista-support/issues/91).*

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
I can add it to the tests at least for coverage then

File: tests/test_widgets.py [hunk 1]
```diff
@@ -197,3 +197,13 @@ def test_widget_sphere():
...
     p.close()
+
+
+@pytest.mark.skipif(NO_PLOTTING, reason="Requires system to support plotting")
+def test_widget_checkbox_button():
+    p = pyvista.Plotter(off_screen=OFF_SCREEN)
+    func = lambda value: value # Does nothing
+    p.add_mesh(mesh)
+    p.add_checkbox_button_widget(callback=func)
+    p.clear_button_widgets()
```
Comment (banesullivan) [inline tests/test_widgets.py]: Let's remove this line. The clear method should be called on close and if not, it will segfault resulting in a failed test

Comment (GuillaumeFavelier) [reply tests/test_widgets.py]: I removed it but I think the same is done in all the other tests. In this case we should correct this

Comment (banesullivan) [reply tests/test_widgets.py]: ah yep, i totally missed that all the other tests are doing that. we should fix those too

**Comment by GuillaumeFavelier** [comment]
I'll go ahead and merge this then I'll open a follow-up PR to fix the tests as suggested in https://github.com/pyvista/pyvista/pull/528#discussion_r365889252.

**Pull request merged**